torchvision/models/detection/_utils.py

```diff
@@ -159,10 +159,8 @@ def encode_single(self, reference_boxes: Tensor, proposals: Tensor) -> Tensor:
         return targets
 
     def decode(self, rel_codes: Tensor, boxes: List[Tensor]) -> Tensor:
-        if not isinstance(boxes, (list, tuple)):
-            raise TypeError(f"This function expects boxes of type list or tuple, instead  got {type(boxes)}")
-        if not isinstance(rel_codes, torch.Tensor):
-            raise TypeError(f"This function expects rel_codes of type torch.Tensor, instead  got {type(rel_codes)}")
+        assert isinstance(boxes, (list, tuple))
+        assert isinstance(rel_codes, torch.Tensor)
```

Comment:
Should we be using `torch._assert()`?

Reply:
you mean everywhere or here in particular? 
I just reverted the changes to what we had previously. https://github.com/pytorch/vision/pull/5587/files#diff-5288f1a12d4cbc77afac42d57b75796da3c8734c0fa4cda80e5c07ac5d16212dL162

Reply:
I meant everywhere in this PR. I understand you are reverting the changes, what's unclear is what is the scope for them. If it's to enable fx-traceability then reverting alone is not enough and you would need to replace asserts with torch._asserts. There is a chance I misunderstand the target of the PR, so might be worth updating the description to explain better the what/why.

Reply:
Yes I was just reverting but you are right as per our offline discussion the entire point is to enable fx-traceability so made the changes accordingly